llvm/LLVM_INVESTIGATION_SUMMARY.md

```diff
@@ -0,0 +1,75 @@
+# Macro Fusion in LLVM for RISC-V
+
+Macro fusion is a performance optimization where specific instruction pairs are combined for more efficient execution, particularly useful in the RISC-V architecture. The LLVM compiler can be customized to support macro fusion through the modification of relevant target description files and by defining fusion rules in the `RISCVMacroFusion.td` file.
+
+For detailed instructions on configuring and enabling macro fusion in LLVM, please refer to the `LLVM_FOR_RISCV.md` file located in the **how-to** repository.
+
+## How Fusions Defined in `.td` Files Get Applied
+
+Fusions defined in `.td` files, like `RISCVMacroFusion.td`, are used to optimize instruction scheduling in the LLVM compilation process. These fusions are integrated through the auto-generated `RISCVGenMacroFusion.inc` and `RISCVGenSubtargetInfo.inc` files, which are built using the `TableGen` tool via the `RISCVCommonTableGen` target. This file contains important functionality for handling various processor-specific features, including macro fusion.
```

Comment:
Fusions is not correct any more than informations is correct.

Use fusion groups or fusion tuples.

Reply:
Done